feat: virtual threads autoconfiguration

chillleader opened the pull request:
## Description

Adds an autoconfiguration to replace the executor service used by Camunda client with a virtual thread executor. This is implemented as a separate module built with Java 21 that can be added instead of / in addition to the normal spring boot starter module.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes #41070

chillleader committed:
feat: virtual threads autoconfiguration

Adds an autoconfiguration to replace the executor service
used by Camunda client with a virtual thread executor.
This is implemented as a separate module built with Java 21
that can be added instead of / in addition to the normal
spring boot starter module.

jonathanlukas commented on clients/camunda-spring-boot-starter-virtual-threads/src/test/java/io/camunda/client/virtualthreads/VirtualThreadsAutoConfigurationIT.java:
```diff
@@ -0,0 +1,82 @@
+
+@SpringBootTest(
+    classes = {VirtualThreadsAutoConfiguration.class, CamundaAutoConfiguration.class},
+    properties = {"camunda.client.mode=selfManaged", "camunda.client.zeebe.enabled=false"})
```
```suggestion
    properties = {"camunda.client.mode=selfManaged", "camunda.client.enabled=false"})
```

jonathanlukas commented on clients/camunda-spring-boot-starter-virtual-threads/pom.xml:
```diff
@@ -0,0 +1,112 @@
+    <dependency>
+      <groupId>io.camunda</groupId>
+      <artifactId>camunda-spring-boot-starter</artifactId>
+    </dependency>
```
```suggestion
      <artifactId>camunda-spring-boot-starter</artifactId>
      <scope>provided</scope>
    </dependency>
```

Using the extension without the actual starter should not work imo.

jonathanlukas reviewed (commented):
I have some comments, please check them :)

chillleader committed:
fix: set camunda-spring-boot-starter scope to provided

chillleader committed:
fix: use the right property syntax in test

chillleader committed:
refactor: MeterRegistry detection & injection

This changes how MeterRegistry injection
is performed, it is now done the same way
here as in CamundaActuatorConfiguration

chillleader committed:
fix: virtual thread factory

The previous approach with a separate counter
is wrong and does not work properly
(counter is initialized only once)

jonathanlukas reviewed (approved):
Thank you!
